### PR TITLE
Add ga tracking

### PR DESCRIPTION
**Pull request by rupertbates**
# Why are you doing this?
To enable server side conversion tracking for Google Analytics - this should make our GA data on conversions and by extension any Optimize test results much more reliable.

I have also added in [enhanced ecommerce tracking](https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#enhancedecom) to enable us to use revenue as an objective in Optimize

## Changes
* Refactor the external API of the library to be the `AcquisitionService` class, this allows clients to submit one request which is then sent to both Ophan and GA.
* This service now returns an `Either` with a list of errors rather than a single error in the failure case.
* Implement GA tracking using the [Measurement Protocol](https://developers.google.com/analytics/devguides/collection/protocol/v1/) - I have included enhanced ecommerce tracking in the tracking data.

### CODE REVIEW

File: src/main/scala/com/gu/acquisition/services/GAService.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  private[services] def buildPayload(submission: AcquisitionSubmission): String = {
+    val transactionId = UUID.randomUUID().toString
+    val body = Map(
+      "v" -> "1", //Version
+      "tid" -> gaPropertyId,
+      "dh" -> submission.gaData.hostname,
+      "uip" -> submission.gaData.clientIp.getOrElse(""), // IP Override
+      "ua" -> submission.gaData.clientUserAgent.getOrElse(""), // User Agent Override
+      "uid" -> submission.ophanIds.browserId.getOrElse(""), // TODO: Or visitId? Does this work here? https://support.google.com/analytics/answer/3123662
```
Comment (rupertbates) [inline src/main/scala/com/gu/acquisition/services/GAService.scala]: Does this seem sensible?

File: src/main/scala/com/gu/acquisition/services/DefaultAcquisitionService.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import scala.concurrent.ExecutionContext
+
+class DefaultAcquisitionService(implicit client: OkHttpClient) extends AcquisitionService {
+  private val ophanService = new OphanService()
+  private val gAService = new GAService()
+  override def submit[A: AcquisitionSubmissionBuilder](a: A)(implicit ec: ExecutionContext) = {
+    val ov = ophanService.submit(a).value
+    val gv = gAService.submit(a).value
+    val result = for {
+      o <- ov
```
Comment (jranks123) [inline src/main/scala/com/gu/acquisition/services/DefaultAcquisitionService.scala:18]: nitpicking but do we need to define these vals outside the `for` comprehension?

**Review by jranks123 — APPROVED** [review]

**Comment by jranks123** [comment]
Looks good, one small comment

File: src/main/scala/com/gu/acquisition/services/DefaultAcquisitionService.scala (hunk 2 above)
Comment (rupertbates) [reply src/main/scala/com/gu/acquisition/services/DefaultAcquisitionService.scala:18]: I believe we need to create them outside the for comprehension in order for them to run in parallel https://alvinalexander.com/scala/how-use-multiple-scala-futures-in-for-comprehension-loop

Comment (jranks123) [reply src/main/scala/com/gu/acquisition/services/DefaultAcquisitionService.scala:18]: Makes sense

**Pull request merged**